### PR TITLE
[Core] Rename PublisherService to SubscriberService

### PR DESCRIPTION
**Pull request by lixin-wei**
## Why are these changes needed?

I think `PublisherClient` is a more reasonable name than `SubscriberClient` since XClient means ‘client used to access X’, like GcsClient.

Besides, in the current codebase we already called this client `publisher_client`(line 329/333), while the actual class name is `SubscriberClient`, this is inconsistent.
https://github.com/ray-project/ray/blob/a8d7897a56d8dc2d2af550adee8cd5399c39dfb9/src/ray/pubsub/subscriber.cc#L326-L339

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Comment by mwtian** [comment]
I agree with the convention of naming GRPC client to XyzService as XyzClient. But the `PublisherService` in `pubsub.proto` seems not the most appropriate name. It seems to imply it is a service for publishing, but actually it is for subscribing. Also the `PublisherService` is not installed anywhere.

If there is a service containing just long polling and subscriber command handlers, it should be named `SubscriberService` and the client should be named accordingly. So if we want to change anything, I think we can just rename the `PublisherService` in `pubsub.proto` to `SubscriberService` or `PubsubService`.

**Comment by lixin-wei** [comment]
@mwtian Sorry for the late response, got some internal business to do last 2 weeks.

I agree with what you said. But I think maybe `SubscribingService` is a better name than `SusbcriberSerivce`. Some people might think `SubscriberService` is a service provided by the subscriber while others think it's a service used by the subscriber. `SubscribingService` won't have this confusion. What do you think?

I changed `PublisherService` to `SubscribingSerivce`, `SubscriberClient` to `SubscribingClient`, and modified all related code I can find, would you please have a look?

**Comment by lixin-wei** [comment]
@mwtian Ping, would you please have a look? :)

**Comment by lixin-wei** [comment]
Changed to SusbcriberSerivce.

File: src/ray/pubsub/test/integration_test.cc [hunk 1]
```diff
@@ -195,7 +195,7 @@ class IntegrationTest : public ::testing::Test {
   rpc::Address address_proto_;
   IOServicePool io_service_ = IOServicePool(3);
   std::unique_ptr<PeriodicalRunner> periodic_runner_;
-  std::unique_ptr<PublisherServiceImpl> publisher_service_;
+  std::unique_ptr<SubscriberServiceImpl> publisher_service_;
```
Comment (mwtian) [inline src/ray/pubsub/test/integration_test.cc:198]: `subscriber_service_`?

File: src/ray/pubsub/test/integration_test.cc [hunk 2]
```diff
@@ -166,7 +166,7 @@ class IntegrationTest : public ::testing::Test {
         /*get_time_ms=*/[]() -> double { return absl::ToUnixMicros(absl::Now()); },
         /*subscriber_timeout_ms=*/absl::ToInt64Microseconds(absl::Seconds(30)),
         /*batch_size=*/100);
-    publisher_service_ = std::make_unique<PublisherServiceImpl>(std::move(publisher));
+    publisher_service_ = std::make_unique<SubscriberServiceImpl>(std::move(publisher));
```
Comment (mwtian) [inline src/ray/pubsub/test/integration_test.cc:169]: `subscriber_service_`?

**Review by mwtian — APPROVED** [review]
Thanks for making the update!

File: src/ray/pubsub/test/integration_test.cc (hunk 1 above)
Comment (lixin-wei) [reply src/ray/pubsub/test/integration_test.cc]: Oh thanks for the catching!

**Comment by lixin-wei** [comment]
@rkooo567 The failed tests seem unrelated, would you please help me merge this PR?

**Comment by lixin-wei** [comment]
@rkooo567  The failed tests seem unrelated, would you please help me merge this PR?

**Pull request merged**